makelib/golang.mk

```diff
@@ -104,7 +104,7 @@ endif
 
 # We use a consistent version of golangci-lint to ensure everyone gets the same
 # linters.
-GOLANGCILINT_VERSION ?= 1.47.1
+GOLANGCILINT_VERSION ?= 1.49.0
```

Comment:
Can we use [`1.50.1`](https://github.com/golangci/golangci-lint/releases/tag/v1.50.1), the latest version? It has more linters made compatible with 1.18 and above.

Reply:
Done. The latest version is `1.49.0` when I submit the PR. I test it as well in my local environment and it works good.